FlowCrypt/src/main/java/com/flowcrypt/email/security/pgp/PgpKey.kt

```diff
@@ -25,42 +25,65 @@ import org.pgpainless.PGPainless
 import org.pgpainless.util.Passphrase
 import java.nio.charset.StandardCharsets
 
+@Suppress("unused")
 object PgpKey {
   fun encryptKey(armored: String, passphrase: String): String {
-    return try {
-      val keys = parseAndNormalizeKeyRings(armored)
-      PGPainless.modifyKeyRing(keys[0] as PGPSecretKeyRing)
-          .changePassphraseFromOldPassphrase(null)
-          .withSecureDefaultSettings()
-          .toNewPassphrase(Passphrase.fromPassword(passphrase))
-          .done()
-          .armor()
-    } catch (e: Exception) {
-      ""
-    }
+    return encryptKey(extractSecretKey(armored), passphrase).armor()
+  }
+
+  private fun encryptKey(key: PGPSecretKeyRing, passphrase: String): PGPSecretKeyRing {
+    return PGPainless.modifyKeyRing(key)
+        .changePassphraseFromOldPassphrase(null)
+        .withSecureDefaultSettings()
+        .toNewPassphrase(Passphrase.fromPassword(passphrase))
+        .done()
   }
 
   fun decryptKey(armored: String, passphrase: String): String {
-    return try {
-      val keys = parseAndNormalizeKeyRings(armored)
-      PGPainless.modifyKeyRing(keys[0] as PGPSecretKeyRing)
-          .changePassphraseFromOldPassphrase(Passphrase.fromPassword(passphrase))
-          .withSecureDefaultSettings()
-          .toNoPassphrase()
-          .done()
-          .armor()
-    } catch (ex: Exception) {
-      ""
-    }
+    return decryptKey(extractSecretKey(armored), passphrase).armor()
   }
 
+  private fun decryptKey(key: PGPSecretKeyRing, passphrase: String): PGPSecretKeyRing {
+    return PGPainless.modifyKeyRing(key)
+        .changePassphraseFromOldPassphrase(Passphrase.fromPassword(passphrase))
+        .withSecureDefaultSettings()
+        .toNoPassphrase()
+        .done()
+  }
+
+  fun changeKeyPassphrase(
+      armored: String,
+      oldPassphrase: String,
+      newPassphrase: String
+  ): String {
+    return changeKeyPassphrase(extractSecretKey(armored), oldPassphrase, newPassphrase).armor()
+  }
+
+  private fun changeKeyPassphrase(
+      key: PGPSecretKeyRing,
+      oldPassphrase: String,
+      newPassphrase: String
+  ): PGPSecretKeyRing {
+    return encryptKey(decryptKey(key, oldPassphrase), newPassphrase)
+  }
+
+  private fun extractSecretKey(armored: String): PGPSecretKeyRing {
+    val key = parseAndNormalizeKeyRings(armored).firstOrNull()
+        ?: throw IllegalArgumentException("Keys not found")
+    if (key is PGPSecretKeyRing)
```

Comment:
@tomholub Sorry to disturb
1) Could you clarify is it the same as we discussed at #1122?

```diff
- if (key is PGPSecretKeyRing)
-      return key
-    else
-     throw IllegalArgumentException("Key is not a secret key")
+ return (key as? PGPSecretKeyRing)?:throw IllegalArgumentException("Key is not a secret key")
```

2) And what about braces to 'if' and 'else' statements?

```diff
- if (key is PGPSecretKeyRing)
-      return key
-    else
-     throw IllegalArgumentException("Key is not a secret key")
+ if (key is PGPSecretKeyRing) {
+    return key
+ } else {
+    throw IllegalArgumentException("Key is not a secret key")
+ }
```

Reply:
if/else statements do not have to have braces **if on one line**. But in that case it's only permitted to use for "consistent" result, eg like a ternary statement `return if (c) 1 else 2` or `val x = if (c) 1 else 2`. Cannot mix return and throw on one line or without braces.

if/else statements **MUST have braces if multiline**. I wish there was some ktlint rule for this (maybe there is?)

```ts
// good
val x = if(c) 1 else 2

// good
return if(c) 1 else 2

// bad
if (key is PGPSecretKeyRing)
      return key
    else
     throw IllegalArgumentException("Key is not a secret key")

// good
if (key is PGPSecretKeyRing) {
  return key
} else {
  throw IllegalArgumentException("Key is not a secret key")
}

// also good but depends on style? I prefer this but other people prefer the above. Not much difference
if (key is PGPSecretKeyRing) {
  return key
}
throw IllegalArgumentException("Key is not a secret key")
```

This is also good. Maybe the best - I find it succinct and clear:

```kt
// good
return (key as? PGPSecretKeyRing) ?: throw IllegalArgumentException("Key is not a secret key")

// if it doesn't fit on one line, a `?: throw` on next line is also good
return (key as? PGPSecretKeyRing)
  ?: throw IllegalArgumentException("Key is not a secret key")
```

I think everybody will have a little different idea, but I think this is a decent compromise.